reference/test/stop/residual_norm_kernels.cpp

```diff
@@ -302,6 +302,134 @@ TYPED_TEST(RelativeResidualNorm, WaitsTillResidualGoalMultipleRHS)
 }
 
 
+template <typename T>
+class ImplicitResidualNormReduction : public ::testing::Test {
+protected:
+    using Mtx = gko::matrix::Dense<T>;
+    using NormVector = gko::matrix::Dense<gko::remove_complex<T>>;
+
+    ImplicitResidualNormReduction()
+    {
+        exec_ = gko::ReferenceExecutor::create();
+        factory_ = gko::stop::ImplicitResidualNormReduction<T>::build()
+                       .with_reduction_factor(r<T>::value)
+                       .on(exec_);
+    }
+
+    std::unique_ptr<
+        typename gko::stop::ImplicitResidualNormReduction<T>::Factory>
+        factory_;
+    std::shared_ptr<const gko::Executor> exec_;
+};
+
+TYPED_TEST_SUITE(ImplicitResidualNormReduction, gko::test::ValueTypes);
+
+
+TYPED_TEST(ImplicitResidualNormReduction, CanCreateFactory)
+{
+    ASSERT_NE(this->factory_, nullptr);
+    ASSERT_EQ(this->factory_->get_parameters().reduction_factor,
+              r<TypeParam>::value);
+    ASSERT_EQ(this->factory_->get_executor(), this->exec_);
+}
+
+
+TYPED_TEST(ImplicitResidualNormReduction, CannotCreateCriterionWithoutB)
+{
+    ASSERT_THROW(this->factory_->generate(nullptr, nullptr, nullptr, nullptr),
+                 gko::NotSupported);
+}
+
+
+TYPED_TEST(ImplicitResidualNormReduction, CanCreateCriterionWithB)
```

Comment:
For this test and the ones above: Do we need them for both reference and omp, or can we remove them from one of these two files?

Reply:
Both are calling different kernels, so I would prefer to have it similar to as it is now for the other stopping criteria.